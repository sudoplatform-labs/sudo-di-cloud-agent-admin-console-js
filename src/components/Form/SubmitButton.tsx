import { Button } from 'antd';
import { ButtonSize } from 'antd/lib/button';
import React from 'react';
import styled from 'styled-components';

const StyledButton = styled(Button)`
  margin-top: 16px;
  margin-bottom: 24px;
  width: 215px;
  font-weight: 600;
`;

interface Props {
  loading?: boolean;
  buttonText: string;
  disabled?: boolean;
  size?: ButtonSize;
  className?: string;
}

/** @deprecated - use <FormButton.Submit /> */
export const SubmitButton = (props: Props): JSX.Element => (
  <StyledButton
    htmlType="submit"
    type="primary"
    shape="round"
    className={props.className}
    size={props.size}
    disabled={props.disabled}
    loading={props.loading}
  >
    {props.buttonText}
  </StyledButton>
);

SubmitButton.defaultProps = {
  size: 'large',
};
