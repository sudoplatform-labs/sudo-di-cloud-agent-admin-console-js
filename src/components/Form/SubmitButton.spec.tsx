import React from 'react';

import { SubmitButton } from './SubmitButton';
import { shallow } from 'enzyme';

describe('SubmitButton', () => {
  it('should render correctly when isBusy is false', () => {
    const component = shallow(
      <SubmitButton loading={false} buttonText="Kiwi" />,
    );

    expect(component).toMatchSnapshot();
  });

  it('should render correctly when isBusy is true', () => {
    const component = shallow(
      <SubmitButton loading={true} buttonText="Kiwi" />,
    );

    expect(component).toMatchSnapshot();
  });
});
