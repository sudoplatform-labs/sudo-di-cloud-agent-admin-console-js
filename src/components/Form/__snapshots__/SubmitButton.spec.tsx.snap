// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly when isBusy is false 1`] = `
<Styled(Button)
  htmlType="submit"
  loading={false}
  shape="round"
  size="large"
  type="primary"
>
  Kiwi
</Styled(Button)>
`;

exports[`SubmitButton should render correctly when isBusy is true 1`] = `
<Styled(Button)
  htmlType="submit"
  loading={true}
  shape="round"
  size="large"
  type="primary"
>
  Kiwi
</Styled(Button)>
`;
